cache/in-memory/src/event/role.rs

```diff
@@ -123,9 +123,11 @@ mod tests {
             assert!(guild_1_role_ids.iter().all(|id| cached_roles.contains(id)));
 
             // Check for the cached role
-            assert!(guild_1_roles
-                .into_iter()
-                .all(|role| cache.role(role.id).expect("Role missing from cache") == role))
+            assert!(guild_1_roles.into_iter().all(|role| cache
+                .role(role.id)
```

Comment:
Can this use `assert_eq!`?

Reply:
No, because a comparison is being done over all the elements of the iterator rather than the iterator itself. This code may need to be refactored but I would prefer to leave that to another PR.